core/domain/wipeout_service.py

```diff
@@ -1253,3 +1256,85 @@ def _pseudonymize_models_transactional(voiceover_application_models):
+    """Pseudonymizes the blog post models for the user with user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending
+            deletion request object to be saved in the datastore.
+    """
+    user_id = pending_deletion_request.user_id
+
+    # We want to preserve the same pseudonymous user ID on all the models
+    # related to one blog post. So we collect all the users' blog
+    # post models and blog post summary models then
+    # we generate a pseudonymous user ID and replace the user ID
+    # with that pseudonymous user ID in all the models.
+    blog_post_model_class = blog_post_models.BlogPostModel
+    blog_post_models_list = blog_post_model_class.query(
+        blog_post_model_class.author_id == user_id
+        ).fetch()
+    blog_post_ids = {model.id for model in blog_post_models_list}
+
+    blog_post_summary_model_class = blog_post_models.BlogPostSummaryModel
+    blog_post_summary_models = blog_post_summary_model_class.query(
+        blog_post_summary_model_class.author_id == user_id
+        ).fetch()
+    blog_post_ids |= {model.id for model in blog_post_summary_models}
+
+    _save_pseudonymizable_entity_mappings_to_different_pseudonyms(
+        pending_deletion_request, models.NAMES.blog_post, blog_post_ids)
+    # TODO(#13053): Add wipeout for BlogPostsRightsModel after adding
+    # service layer.
+    @transaction_services.run_in_transaction_wrapper
+    def _pseudonymize_models_transactional(
+            blog_posts_related_models, pseudonymized_id):
+        """Pseudonymize user ID fields in the models.
+
+        This function is run in a transaction, with the maximum number of
+        blog_posts_related_models being MAX_NUMBER_OF_OPS_IN_TRANSACTION.
+
+        Args:
+            blog_posts_related_models: list(BaseModel). Models whose user IDs
+                should be pseudonymized.
+            pseudonymized_id: str. New pseudonymized user ID to be used for
+                the models.
+        """
+        blog_post_models_list = [
+            model for model in blog_posts_related_models
+            if isinstance(model, blog_post_model_class)]
+        for blog_post_model in blog_post_models_list:
+            if blog_post_model.author_id == user_id:
+                blog_post_model.author_id = pseudonymized_id
+            blog_post_model.update_timestamps()
+
+        blog_post_summary_models_list = [
+            model for model in blog_posts_related_models
+            if isinstance(model, blog_post_summary_model_class)]
+        for blog_post_summary in blog_post_summary_models_list:
+            if blog_post_summary.author_id == user_id:
+                blog_post_summary.author_id = pseudonymized_id
+            blog_post_summary.update_timestamps()
+
+        datastore_services.put_multi(
+            blog_post_models_list +
+            blog_post_summary_models_list)
+
+    blog_posts_ids_to_pids = (
```

Comment:
blog_post_ids_to_pids

(Note: if you say "blog posts ids" then the idea here is that a single ID refers to multiple blog posts. A "blog post id" is a thing, and then you pluralize that.)

Reply:
Fixed.